app/lib/actions/katello/repository/check_matching_content.rb

```diff
@@ -62,6 +63,12 @@ def package_groups_match?(source_repo, target_repo)
         def distributions_match?(source_repo, target_repo)
           source_repo.distribution_information == target_repo.distribution_information
         end
+
+        def yum_metadata_files_match?(source_repo, target_repo)
+          source_repo_checkums = source_repo.yum_metadata_files.pluck(:checksum).sort
+          target_repo_checkums = target_repo.yum_metadata_files.pluck(:checksum).sort
```

Comment:
I think theoretically there could still be a problem if the filename changed but the checksum didn't?   Is it possible to handle this situation too?